libs/wingc/src/lsp/document_symbols.rs

```diff
@@ -1,13 +1,154 @@
-use crate::ast::{Scope, *};
-use crate::lsp::ast_traversal::{
-	get_expressions_from_statement, get_statements_from_scope_no_context, get_symbols_from_expression,
-	get_symbols_from_statement, TreeLocationContext,
-};
+use crate::ast::*;
 use crate::lsp::sync::FILES;
+use crate::visit::Visit;
 use crate::wasm_util::{ptr_to_string, string_to_combined_ptr, WASM_RETURN_ERROR};
 use lsp_types::DocumentSymbol;
 use lsp_types::SymbolKind;
 
+pub struct DocumentSymbolVisitor {
+	pub document_symbols: Vec<DocumentSymbol>,
+}
+
+impl DocumentSymbolVisitor {
+	pub fn new() -> Self {
+		Self {
+			document_symbols: vec![],
+		}
+	}
+}
+
+impl Visit<'_> for DocumentSymbolVisitor {
+	fn visit_stmt(&mut self, statement: &Stmt) {
```

Comment:
Cool use of the visitor 👍 Curious if based on the name it's possible to use the visit_symbol method you added, or if there's a reason it wasn't used here?

Reply:
DocumentSymbol is more concerned about how symbols are introduced rather than every usage. Currently, I decided to only handle cases with statements, which are probably the most common way this happens right now.